src/App.tsx

```diff
@@ -1,24 +1,95 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { SelectStatus } from './types/SelectStatus';
+import { TodoError } from './types/TodoError';
+import { ErrorTab } from './components/ErrorTab';
+import * as todoService from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedStatus, setSelectedStatus] = useState(SelectStatus.All);
+  const [errorMesage, setErrorMesage] = useState(TodoError.empty);
+  const [newAddedTodoId, setNewAddedTodoId] = useState<number | null>(null);
+
+  const getFilteredTodos = (todos: Todo[]) => {
+    const filteredTodos = [...todos];
+
+    switch (selectedStatus) {
+      case SelectStatus.Active:
+        return filteredTodos.filter(todo => !todo.completed);
+
+      case SelectStatus.Completed:
+        return filteredTodos.filter(todo => todo.completed);
+
+      default:
+        return filteredTodos;
+    }
```

Comment:
No need to create copy - filter method do not mutate original array
```suggestion
    switch (selectedStatus) {
      case SelectStatus.Active:
        return todos.filter(todo => !todo.completed);

      case SelectStatus.Completed:
        return todos.filter(todo => todo.completed);

      default:
        return todos;
    }
```